core/discovery/storage_test.go

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright (C) 2019 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package discovery
+
+import (
+	"testing"
+
+	"github.com/mysteriumnetwork/node/core/discovery/reducer"
+	"github.com/mysteriumnetwork/node/market"
+	"github.com/mysteriumnetwork/node/market/mysterium"
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	proposalProvider1Streaming = market.ServiceProposal{
+		ServiceType: "streaming",
+		ProviderID:  "0x1",
+	}
+	proposalProvider1Noop = market.ServiceProposal{
+		ServiceType: "noop",
+		ProviderID:  "0x1",
+	}
+	proposalProvider2Streaming = market.ServiceProposal{
+		ServiceType: "streaming",
+		ProviderID:  "0x2",
+	}
+)
+
+type filter struct {
+	serviceType string
+}
+
+func (filter *filter) Matches(proposal market.ServiceProposal) bool {
+	return filter.serviceType == "" || proposal.ServiceType == filter.serviceType
+}
+
+func (filter *filter) ToAPIQuery() mysterium.ProposalsQuery {
+	return mysterium.ProposalsQuery{
+		ServiceType: filter.serviceType,
+	}
+}
+
+func Test_Storage_List(t *testing.T) {
+	storage := NewStorage()
+	assert.Equal(t, []market.ServiceProposal{}, storage.Proposals())
+
+	storage = mockStorage()
```

Comment:
mockStorage name is confusing and sounds like you testing some mock storage but here you are testing actual storage implementation with some test proposals. Maybe something like

```go
func newProposalStorage(proposals ...[]market.ServiceProposal) *ProposalStorage {
	return &ProposalStorage{
		proposals: proposals...,
	}
}

// in tests
storage = newProposalStorage()
// assert
...
storage = newProposalStorage(proposalProvider1Streaming, proposalProvider1Noop, proposalProvider2Streaming)
// assert

Reply:
Improved clarity of mocking